packages/cli/src/ceramic-logger-plugins.ts

```diff
@@ -86,28 +113,125 @@ export class LogToFiles {
 
             if (match !== null) {
                 const docId = match[0]
-                const filePath = filePrefix + '-docids.log'
-                LogToFiles._writeStream(filePath, docId, 'w')
+                const filePath = `${basePath}${filePrefix}-docids.log`
+                const result = await LogToFiles._writeStream(blockedFiles, filePath, docId, 'w')
+                if (result.blocked) await LogToFiles._writeBlockedWarning(blockedFiles, basePath, filePath)
             }
         }
     }
 
     /**
-     * Returns true if it has been `MINUTES_TO_EXPIRATION` minutes since the file was created
-     * @notice Returns false if file is not found.
-     * Returns true if unable to get file creation datetime for some other reason.
+     * Logs and writes warning message about `filePath` being blocked.
+     * @notice If the file that this tries to write to itself is blocked,
+     * its message will be dropped and only appear in the console
+     * @param basePath Base directory for file to write to
+     * @param filePath Path to file that is blocked
+     */
+    private static async _writeBlockedWarning (
+        blockedFiles: BlockedFiles,
+        basePath: string,
+        filePath: string
+    ): Promise<void> {
+        const loggerFilePath = basePath + 'logger.log'
+        const message = `Stream blocked for file: ${filePath}\n`
+        await LogToFiles._writeStream(blockedFiles, loggerFilePath, message, 'a')
+    }
+
+    /**
+      * Writes `message` to a file and returns whether it is blocked or not
+      * @notice If the file is currently being written to by an open stream or if
+      * its buffer is full, the call to this method will return without writing
+      * @param filePath Full path of file to write to
+      * @param message Message to write to `filePath`
+      * @param writeFlag Specifies writing method (e.g. "a" for append, "w" for overwrite)
+      * @returns True if the stream is blocked and false otherwise
+      */
+    private static async _writeStream (
+        blockedFiles: BlockedFiles,
+        filePath: string,
+        message: string,
+        writeFlag: string
+    ): Promise<WriteResult> {
+        if (blockedFiles[filePath]) {
+            console.warn(`Stream busy for ${filePath}. Some logs may be dropped.`)
+            return { blocked: blockedFiles[filePath], filePath }
+        }
+
+        blockedFiles[filePath] = true
+
+        const stream = fs.createWriteStream(filePath, { flags: writeFlag })
```

Comment:
Hmm... I don't see a reference to 'fs' in our package.json file.  Do you know which specific package this is we're using?  I'm trying to find docs as to whether `fs.createWriteStream` creates a new file descriptor each time, or if it keeps open a longer lived file handle under the hood.

I think this most likely creates and destroys a fd every time we log a line, which may cause us some perf issues down the road, but doing the work to keep a file handle alive longer is non-trivial and probably not worth it right now.  Just something to keep an eye on going forward as a place for future optimization.

Reply:
Fs is native to Node.js. It creates a new fd each time, so yes it does create and destroy, which we want for now to ensure we don't have fds lying around open.

Reply:
For context: https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options

Reply:
Unfortunately for now I haven't found a way around this. At least working with docker, the attached volume doesn't get the writes until the __fd is closed__. So even when I use `stream.cork`, `stream.uncork` which is _supposed_ to buffer and flush, docker does see anything until fd is closed/process is destroyed.